### PR TITLE
fix: update last_instant when rotating into persistable window

### PR DESCRIPTION
**Pull request by tustvold**
There is a bug in #2054 where when windows were rotated into the persistable window it wouldn't update the end instant of the persistable window. This would result in windows with incorrect reported time_of_last_write

### CODE REVIEW

File: persistence_windows/src/persistence_windows.rs [hunk 1]
```diff
@@ -499,6 +499,10 @@ impl Window {
 
     /// Add one window to another. Used to collapse closed windows into persisted.
     fn add_window(&mut self, other: Self) {
+        assert!(self.last_instant <= other.created_at);
+        assert!(self.last_instant <= other.last_instant);
```
Comment (alamb) [inline persistence_windows/src/persistence_windows.rs:503]: thank you for including these asserts.

**Review by e-dard — APPROVED** [review]
👍

File: persistence_windows/src/persistence_windows.rs [hunk 2]
```diff
@@ -1407,6 +1412,34 @@ mod tests {
                     row_count: 8
                 },
             ]
-        )
+        );
+
+        // Rotate first and second windows into persistable
+        w.rotate(instant + late_arrival_period + DEFAULT_CLOSED_WINDOW_PERIOD * 2);
```
Comment (alamb) [inline persistence_windows/src/persistence_windows.rs:1418]: how does this test the update of `last_instant`? Does that end up in `time_of_last_write` with the value in window 2?

**Review by alamb — APPROVED** [review]

File: persistence_windows/src/persistence_windows.rs (hunk 2 above)
Comment (tustvold) [reply persistence_windows/src/persistence_windows.rs:1418]: Correct :+1:

**Pull request merged**